libraries/AP_HAL/board/linux.h

```diff
@@ -211,6 +211,21 @@
 #define HAL_INS_DEFAULT HAL_INS_HIL
 #define HAL_BARO_DEFAULT HAL_BARO_HIL
 #define HAL_COMPASS_DEFAULT HAL_COMPASS_HIL
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_OCPOC_ZYNQ
+#define HAL_BOARD_LOG_DIRECTORY "/var/APM/logs"
+#define HAL_BOARD_TERRAIN_DIRECTORY "/var/APM/terrain"
+#define HAL_INS_DEFAULT HAL_INS_MPU9250_SPI
+#define HAL_INS_DEFAULT_ROTATION ROTATION_NONE
+#define HAL_INS_MPU9250_NAME "mpu9250"
+#define HAL_BARO_DEFAULT HAL_BARO_MS5611_SPI
+#define HAL_BARO_MS5611_NAME "ms5611"
+#define HAL_COMPASS_DEFAULT HAL_COMPASS_OCPOC_ZYNQ
+#define OCPOC_DEFAULT_RCIN_SBUS 1
```

Comment:
Why do you add this define here? In the Linux class you already ifdef for your board, so I don't see the point.

Reply:
This is a bit of an oddity, but currently we have two separate 3-pin GPIOs dedicated for RC Input, one for SBUS and one for PPM. The GPIOs are routed through the OcPoC's FPGA, and the SBUS and PPM input require separate IP cores, so we have to set here whether or not ardupilot should read SBUS or PPM. I know it's a little clunky, but it works and it's well tested.

We have a plan to eventually try accepting SBUS and PPM on the same GPIO, but that will take some work on the FPGA side, additional development in the RCInput driver, and a good bit of testing to call it good.

Reply:
But that means it's a compile-time option only.  I'll accept it as is for now. We can work on improving it so we accept sbus and ppm at the same time.  If you have a spare gpio pin (or the correspondent sbus IP on the fpga can also be configurable over i2c/spi/other), it would also be possible to support DSM and other RC protocols that are basically a UART.

Are these fpga IPs also open source?  We have [some on Aero](https://github.com/intel-aero/intel-aero-fpga), including the I2C part if you are interested. We would also probably be interested in borrowing some of yours if they are open source.

Reply:
Currently the IPs are not open source. The RC Receiver is an in-house AXI compatible design, and the SBUS is an AXI_UART16550A with the TX not exposed and the RX inverted.

We have a few different options that we can try to use to make this work, the simplest, tying the input interface to the same pin, and watching the data input between the two.